src/Discord.Net.Rest/Entities/Channels/RestTextChannel.cs

```diff
@@ -168,5 +173,13 @@ IAsyncEnumerable<IReadOnlyCollection<IUser>> IChannel.GetUsersAsync(CacheMode mo
             else
                 return AsyncEnumerable.Empty<IReadOnlyCollection<IGuildUser>>();
         }
+
+        // INestedChannel
+        async Task<ICategoryChannel> INestedChannel.GetCategoryAsync(CacheMode mode, RequestOptions options)
+        {
```

Comment:
I believe this would be a better way (to not rely on the Guild on Rest):
```cs
if (mode == CacheMode.AllowDownload)
    return await GetCategoryAsync(options).ConfigureAwait(false);
else
    return null;
```

That's what I meant with "follow the same pattern as other rest entities and call the inner `GetCategoryAsync`" in my last review.
See:https://github.com/RogueException/Discord.Net/blob/dev/src/Discord.Net.Rest/Entities/Channels/RestTextChannel.cs#L141

Reply:
This is implemented, but isn't showing as outdated in Github.